File: examples/java-and-scala/src/scala/com/systemvi/java/Interface.java
package com.systemvi.java;

interface Drawable {
    void draw();
}
interface Disposable{
    void dispose();
}
interface Player{
    void play();
    void pause();
    void restart();
}

class Test implements Drawable, Disposable,Player{
    public Test(){
        //load resources 
    }

    @Override
    public void dispose() {
        //free resources
    }

    @Override
    public void play() {
        //start something
    }

    @Override
    public void pause() {
        //pause something
    }

    @Override
    public void restart() {
        
    }

    @Override
    public void draw() {
        
    }

    public static void main(String[] args) {
        Test test = new Test();
        Drawable drawable = test;
        Player player = test;
        Disposable disposable = test;
    }
}

File: examples/java-and-scala/src/scala/com/systemvi/scala/Interface.scala
package com.systemvi.scala


//interface se zove trait
trait Init{
  def init(): Unit
}
trait Disposable {
  def dispose():Unit
}
trait Drawable{
  def draw():Unit
}

//za prvi trait se pise extends, za svaki sledeci with

class Test extends Disposable with Drawable with Init{

  //posto je celo telo klase konstruktor ovde mosemo da pozovemo init
  init()
  
  override def init(): Unit = {
    
  }
  override def draw(): Unit = {
    
  }
  override def dispose(): Unit = {
    
  }
}
